astropy/stats/circstats.py

```diff
@@ -0,0 +1,447 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+"""
+This module contains simple functions for dealing with circular statistics, for
+instance, mean, variance, standard deviation, correlation coefficient, and so
+on. This module also cover tests of uniformity, e.g., the Rayleigh and V tests.
+The Maximum Likelihood Estimator for the Von Mises distribution along with the
+Cramer-Rao Lower Bounds are also implemented. Almost all of the implementations
+are based on reference [1], which is also the basis for the R package
+'CircStats' [2].
+"""
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import numpy as np
+from astropy.units import Quantity
+from astropy import units as u
+from astropy.utils.compat.numpy import broadcast_to
+
+__all__ = ['circmean', 'circvar', 'circmoment', 'circcorrcoef', 'rayleightest',
+           'vtest', 'vonmisesmle']
+__doctest_requires__ = {'vtest': ['scipy.stats']}
+
+
+def _components(data, p=1, phi=0.0, axis=None, weights=None):
+    # Utility function for computing the generalized rectangular components
+    # of the circular data.
+    if weights is None:
+        weights = np.ones((1,))
+    try:
+        weights = broadcast_to(weights, data.shape)
+    except ValueError:
+        raise ValueError('Weights and data have inconsistent shape.')
+
+    C = np.sum(weights * np.cos(p * (data - phi)), axis)/np.sum(weights, axis)
+    S = np.sum(weights * np.sin(p * (data - phi)), axis)/np.sum(weights, axis)
+
+    return C, S
+
+
+def _angle(data, p=1, phi=0.0, axis=None, weights=None):
+    # Utility function for computing the generalized sample mean angle
+    C, S = _components(data, p, phi, axis, weights)
+
+    # theta will be an angle in the interval [-np.pi, np.pi)
+    # [-180, 180)*u.deg in case data is a Quantity
+    theta = np.arctan2(S, C)
+
+    if isinstance(data, Quantity):
+        theta = theta.to(data.unit)
+
+    return theta
+
+
+def _length(data, p=1, phi=0.0, axis=None, weights=None):
+    # Utility function for computing the generalized sample length
+    C, S = _components(data, p, phi, axis, weights)
+    return np.hypot(S, C)
+
+
+def circmean(data, axis=None, weights=None):
+    """ Computes the circular mean angle of an array of circular data.
+
+    Parameters
+    ----------
+    data : numpy.ndarray
+        Array of circular (directional) data.
+    axis : int, optional
+        Axis along which circular means are computed. The default is to compute
+        the mean of the flattened array.
+    weights : numpy.ndarray, optional
+        In case of grouped data, the i-th element of ``w`` represents a
+        weighting factor for each group such that sum(w,axis) equals the number
+        of observations. See [1], remark 1.4, page 22, for detailed
+        explanation.
+
+    Returns
+    -------
+    circmean : Quantity
```

Comment:
If an numpy.ndarray, it returns a numpy.ndarray not a quanity which is okay, but this docstring needs to be updated